ui/dashboards/src/components/AddPanel/AddPanel.tsx

```diff
@@ -41,26 +44,42 @@ const AddPanel = ({ isOpen, onClose }: AddPanelProps) => {
   const [group, setGroup] = useState(0);
   const [panelName, setPanelName] = useState('');
   const [panelDescription, setPanelDescription] = useState('');
+  const [kind, setKind] = useState('');
+  const [options, setOptions] = useState<JsonObject>({});
 
-  const onSelectGroupChange = (e: SelectChangeEvent<number>) => {
-    setGroup(e.target.value as number);
+  const handleGroupChange: SelectProps<number>['onChange'] = (e) => {
+    const { value } = e.target;
+
+    // Handle string (which would be empty string but shouldn't happen since we don't allow a "None" option) by
+    // just ignoring it
+    if (typeof value === 'string') return;
+
+    setGroup(value);
   };
 
-  const onPanelNameChange = (e: ChangeEvent<HTMLInputElement>) => {
+  const handlePanelNameChange = (e: ChangeEvent<HTMLInputElement>) => {
     setPanelName(e.target.value);
   };
 
-  const onPanelDescriptionChange = (e: ChangeEvent<HTMLInputElement>) => {
+  const handlePanelDescriptionChange = (e: ChangeEvent<HTMLInputElement>) => {
     setPanelDescription(e.target.value);
   };
 
+  const handleKindChange: SelectProps<string>['onChange'] = (e) => {
+    setKind(e.target.value);
+  };
+
+  const handleOptionsChange: PanelOptionsEditorProps['onChange'] = (next) => {
+    setOptions(next);
+  };
+
   const onAddPanelClick = (e: FormEvent) => {
     e.preventDefault();
     const panelKey = removeWhiteSpacesAndSpecialCharacters(panelName);
     addPanel(panelKey, {
-      kind: 'EmptyChart',
+      kind,
```

Comment:
More of a UX question, but when adding a new panel, instead of leaving 'Panel Type' empty should it automatically select 'Line Chart' since that's used most often?

Reply:
Seems reasonable to me!

Reply:
Yeah, once those options are dynamic (i.e. we have to ask the plugin system what panels are available), I think it's an open question what the default should be. In theory it's possible for someone to use this without the `LineChart` plugin.